### PR TITLE
Added documentation for utils.php

### LINKED ISSUE #61: Add code documentation
Please document the new functions.

* What do they do?
* What parameters do they expect?
* What do they return?

### PR DESCRIPTION
**Pull request by Mipronimo**
The documentation added for utils.php
closes #61

### CODE REVIEW

File: lib/Auth/utils.php [hunk 1]
```diff
@@ -86,6 +98,20 @@ public function fetchAuthToken($serverconfig) {
 		return $token;
 	}
 
+	/**
+	 * This function can edit the state to enter the needed token types for a user.
+	 * If the user needs u2f, the u2f sign request will be triggered.
+	 * If the user needs an otp, the otp input will appear in the login form.
+	 *
+	 * @param $state
+	 * The state is needed to be changed in this function
```
Comment (cornelinux) [inline lib/Auth/utils.php]: Type of the state?

File: lib/Auth/utils.php [hunk 2]
```diff
@@ -69,6 +69,18 @@ public function curl($params, $headers, $serverconfig, $api_endpoint, $http_meth
 		return $body;
 	}
 
+
+	/**
+	 * With this function you can get the authorization token with a service account.
+	 *
+	 * @param $serverconfig
```
Comment (cornelinux) [inline lib/Auth/utils.php]: Please specify the type of the $serviceconfig. (Being an object, key / value whatever)
Is there a defined way to do this?

File: lib/Auth/utils.php [hunk 3]
```diff
@@ -86,6 +98,20 @@ public function fetchAuthToken($serverconfig) {
 		return $token;
 	}
 
+	/**
+	 * This function can edit the state to enter the needed token types for a user.
+	 * If the user needs u2f, the u2f sign request will be triggered.
+	 * If the user needs an otp, the otp input will appear in the login form.
+	 *
+	 * @param $state
```
Comment (cornelinux) [inline lib/Auth/utils.php]: Type of state?

File: lib/Auth/utils.php [hunk 4]
```diff
@@ -86,6 +98,20 @@ public function fetchAuthToken($serverconfig) {
...
+	 * The state is needed to be changed in this function
+	 *
+	 * @param $body
```
Comment (cornelinux) [inline lib/Auth/utils.php]: type of body?

File: lib/Auth/utils.php [hunk 5]
```diff
@@ -86,6 +98,20 @@ public function fetchAuthToken($serverconfig) {
 		return $token;
 	}
 
+	/**
+	 * This function can edit the state to enter the needed token types for a user.
+	 * If the user needs u2f, the u2f sign request will be triggered.
```
Comment (cornelinux) [inline lib/Auth/utils.php]: This does not happen in this function, so I would not mention it here.

This function simply adds new entries to the $state object, depending on the $body. Right?
So simply mention, which keys are added in the $state.

File: lib/Auth/utils.php (hunk 2 above)
Comment (Mipronimo) [reply lib/Auth/utils.php]: fixed in dae2c98

File: lib/Auth/utils.php (hunk 3 above)
Comment (Mipronimo) [reply lib/Auth/utils.php]: fixed in dae2c98

File: lib/Auth/utils.php (hunk 4 above)
Comment (Mipronimo) [reply lib/Auth/utils.php]: fixed in dae2c98

File: lib/Auth/utils.php (hunk 5 above)
Comment (Mipronimo) [reply lib/Auth/utils.php]: fixed in dae2c98

**Pull request merged**